test/test_testing.py

```diff
@@ -4380,37 +4426,40 @@ def test_make_dir_with_permission_mask(self):
             -1,      # Less than 0o000
         )):
             with pytest.raises(pebble.PathError) as excinfo:
-                client.make_dir(f"{self.prefix}/dir3_{i}", permissions=bad_permission)
+                client.make_dir(f"{pebble_dir}/dir3_{i}", permissions=bad_permission)
             assert excinfo.value.kind == 'generic-file-error'
 
-    def test_remove_path(self):
-        client = self.client
-        client.push(f"{self.prefix}/file", '')
-        client.make_dir(f"{self.prefix}/dir/subdir", make_parents=True)
-        client.push(f"{self.prefix}/dir/subdir/file1", '')
-        client.push(f"{self.prefix}/dir/subdir/file2", '')
-        client.push(f"{self.prefix}/dir/subdir/file3", '')
-        client.make_dir(f"{self.prefix}/empty_dir")
+    def test_remove_path(
+        self,
+        pebble_dir: str,
+        client: typing.Union[_TestingPebbleClient, pebble.Client],
+    ):
+        client.push(f"{pebble_dir}/file", '')
```

Comment:
This line fails in CI with `client` being `None`.
I think the containing class is missing a fixture, though that makes me wonder how pytest fulfils the missing argument/fixture 😖 

A somewhat related thought: this file is really long and needs to be split up. Perhaps doing so would actually help to reason about the fixtures.

Reply:
I have found the error and fixed it, thanks for pointing out!

In fact, multiple test files (even `ops/*.py`) could benefit from a split. Shall we do it in another PR to refactor this?

Reply:
My +1 is for another PR to split tests up.